File: E-Commerce/app/src/main/java/com/example/e_commerceapp/models/Vendor.kt
package com.example.ead_pasindu.models
// Vendor.kt
data class Vendor(
    val id: String,
    val name: String,
    val rating: Float
)

data class VendorComment(
    val comment: String
)

data class VendorResponse(
    val success: Boolean,
    val message: String
)

File: EAD_Backend/EAD_Backend/Models/Vendor.cs
/*
 * File name : Vendor.cs
 * Author - Tissera H.M.V.
 * Description - Vendor Model
*/

using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;
using System.Text.Json.Serialization;

namespace EAD_Backend.Models
{
    public class Vendor
    {
        [BsonId]
        [JsonIgnore]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string Name { get; set; }

        public string Rating { get; set; }

        public string Comment { get; set; }

        [JsonIgnore]
        public string CustomerEmail { get; set; }
    }
}
